hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/input/FileInputFormat.java

```diff
@@ -364,13 +364,29 @@ protected void addInputPathRecursively(List<FileStatus> result,
         if (stat.isDirectory()) {
           addInputPathRecursively(result, fs, stat.getPath(), inputFilter);
         } else {
-          result.add(stat);
+          result.add(shrinkStatus(stat));
         }
       }
     }
   }
-  
-  
+
+  public static FileStatus shrinkStatus(FileStatus origStat) {
```

Comment:
This is very much a user-facing class, and all public methods in this file have Javadoc comments. I think it's appropriate here as well.  There should be a comment explaining what this is doing and why. When I first glanced at the code, it looked like it was just making a copy of everything, and it didn't look like it was shrinking anything at all.

Reply:
Thanks for reviewing this!  Comments are added to clarify the changes.
--